src/source-map.ts

```diff
@@ -72,6 +72,7 @@ function createJsIdentitySourcemap(
     generator.addMapping(mapping);
   });
 
+  generator.setSourceContent(sourceUrl, sourceContent);
```

Comment:
Looks like there's cases where there is a newline char (`\n`) at the beginning of the `sourceContent` here that causes the resulting map to be off by one line. I don't know of a simple check for `if (sourceContent.startsWith('\n'))` would be adequate? Might generate false positives.